nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/db/TableSchema.java

```diff
@@ -136,14 +140,23 @@ public static TableSchema from(final Connection conn, final String catalog, fina
             } else {
                 // Parse the Update Keys field and normalize the column names
                 for (final String updateKey : updateKeys.split(",")) {
-                    primaryKeyColumns.add(ColumnDescription.normalizeColumnName(updateKey.trim(), translateColumnNames));
+                    final String colName = normalizedName(updateKey, translateColumnNames, normalizer);
+                    primaryKeyColumns.add(colName);
+
                 }
             }
 
-            return new TableSchema(catalog, schema, tableName, cols, translateColumnNames, primaryKeyColumns, dmd.getIdentifierQuoteString());
+            return new TableSchema(catalog, schema, tableName, cols, translateColumnNames, normalizer, primaryKeyColumns, dmd.getIdentifierQuoteString());
         }
     }
 
+    public static String normalizedName(final String name, final boolean translateColumnNames, final ColumnNameNormalizer normalizer) {
+        final String colName = name.trim();
+        if (translateColumnNames && normalizer != null)
+            return normalizer.getNormalizedName(colName);
```

Comment:
I feel it would be better to put the toUpperCase() here instead of repeating it in every normalizer. 
trim() is not repeated in the normalizers and toUpperCase feels similar to trim.
Also move trim() inside the IF check. translateNames == false should make zero modifications to the names.
return normalizer.getNormalizedName(colName).trim().toUpperCase;